### PR TITLE
Fix my issues and changes

### PR DESCRIPTION
**Pull request by ahmedhnewa**
I have used gal instead of image_gallery_saver

I didn't change the packages and plugin versions

Please change it to fit your needs

[Code documentation and a new event for confirm removal of the image](https://github.com/singerdmx/flutter-quill/pull/1414)
[Improvements and one new event](https://github.com/singerdmx/flutter-quill/pull/1413)

### CODE REVIEW

**Comment by singerdmx** [comment]
You need to rebase

**Comment by singerdmx** [comment]
@Jon-Salmon please have a look

File: flutter_quill_extensions/lib/flutter_quill_extensions.dart [hunk 1]
```diff
@@ -72,21 +72,8 @@ class FlutterQuillEmbeds {
   }) =>
       [
         ImageEmbedBuilder(
-          afterRemoveImageFromEditor: afterRemoveImageFromEditor ??
-              (imageFile) async {
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/lib/flutter_quill_extensions.dart:76]: I was mainly referring only to `shouldRemoveImageFromEdtior` about it being optional. There is nothing wrong with this being optional too, but we should still provide this sensible default here.

**Review by Jon-Salmon — CHANGES_REQUESTED** [review]
Thanks for the updates! Just one small nit, but happy for this to be merged after changing that one small bit.

**Comment by ahmedhnewa** [comment]
> Thanks for the updates! Just one small nit, but happy for this to be merged after changing that one small bit.

Thank you for your concern about this great plugin, I think we need to change the documentation of some of the changed code

I mentioned that the image by default will be removed from the local storage if it is removed from the editor from Quill Delta

But now it will not do anything by default

Instead, in the documentation, we should only show the developer who uses this package how to do it himself since it's not by default

I also suggest changing the names of the parameters that I have added

I mean, it could be shorter and more descriptive, but I was lazy, to be honest, since I already mentioned that I didn't have enough time to make more changes

If you want me to make these changes by myself, please tell me to do it

Or if you can handle it and you think you will do it better, then be my guest.

Thank you for your time and efforts again to the community.

File: flutter_quill_extensions/lib/flutter_quill_extensions.dart (hunk 1 above)
Comment (ahmedhnewa) [reply flutter_quill_extensions/lib/flutter_quill_extensions.dart:76]: > I was mainly referring only to `shouldRemoveImageFromEdtior` about it being optional. There is nothing wrong with this being optional too, but we should still provide this sensible default here.

I agree with you, but then we shouldn't change the documentation code

As I suggest in my previous comment

And I was mainly refering to afterRemoveImageFromEditor

**Comment by singerdmx** [comment]
LGTM

**Comment by Jon-Salmon** [comment]
Thanks for the contributions. I'll make some documentation changes and release a new version shortly

**Pull request merged**